neotron-bmc-pico/src/main.rs

```diff
@@ -510,6 +526,46 @@ mod app {
 					panic!("Q full!");
 				}
 			}
+
+			// The speaker PWM needs to be updated (register was updated)
+			if register_state.speaker_needs_update {
+				defmt::info!("speaker PWM update");
+				register_state.speaker_needs_update = false;
+
+				let config = &mut register_state.speaker_config;
+				let duration = config.duration();
+
+				if duration > 0 {
+					// a note has to be played - enable PWM
+					ctx.shared.speaker_pwm_tim.lock(|tim14| {
+						config.setup_ch(tim14);
+						pwm::Config::enable_ch(tim14);
+					});
+					if let Some(h) = ctx.local.speaker_task_handle.take() {
+						// if there's a running "stop" task, reschedule it
+						defmt::info!("Task reschedule!");
+						match h.reschedule_after((duration as u64).millis()) {
```

Comment:
This `reschedule_after` does a `warn!` on error, but the `spawn_after` below does an `unwrap()`. What's the reason for the difference?